packages/relay/src/formatters.ts

```diff
@@ -265,6 +266,33 @@ const nanOrNumberTo0x = (input: number | BigNumber | bigint | null): string => {
   return input == null || Number.isNaN(input) ? numberTo0x(0) : numberTo0x(input);
 };
 
+const nanOrNumberInt64To0x = (input: number | BigNumber | bigint | null): string => {
+  if (input && input < 0) {
+    // the hex of a negative number can be obtained from the binary value of that number positive value
+    // the binary value needs to be negated and then to be incremented by 1
+
+    // how the transformation works (using 16 bits)
+    // a 16 bits integer variables have values from -32768 to +32767, so:
+    // 0      - 0x0000 - 0000 0000 0000 0000
+    // 32767  - 0x7fff - 0111 1111 1111 1111
+    // -32768 - 0x8000 - 1000 0000 0000 0000
+    // -1     - 0xffff - 1111 1111 1111 1111
+
+    // converting int16 -10 will be done as following:
+    // - make it positive = 10
+    // - 16 bits binary value of 10 = 0000 0000 0000 1010
+    // - inverse the bytes = 1111 1111 1111 0101
+    // - adding +1 = 1111 1111 1111 0110
+    // - 1111 1111 1111 0110 bits = 0xfff6
+
+    // we're using 64 bits integer because that the type returned by the mirror node - int64
+    const bits = 64;
+    return numberTo0x((BigInt(input.toString()) + (BigInt(1) << BigInt(bits))) % (BigInt(1) << BigInt(bits)));
```

Comment:
question: do we expect `number` bigger than `53` bits? so it's not automatically coerced to double https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

Reply:
Yep, I don't know how often it will happen but we definitely must handle values bigger than `Number.MAX_SAFE_INTEGER`. 

The max Int64 positive value is `0x7fffffffffffffff` which has 63 "usable" bits. 
```
0111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111
```

That's why we always convert the incoming value to BigInt here :point_down: 
```typescript
BigInt(input.toString())
```

Hope that makes sense or am I wrong?

Reply:
Mmm, I'm not sure that `BigInt(input.toString())` works as expected when `input` being a `number` uses more than `53` bits. Say

```js
> const x = 0x7fffffffffffffff
> typeof(x)
'number'
> BigInt(x.toString()) - 0x7fffffffffffffffn
193n
```

but the difference should be zero, right?

Reply:
You're right, the difference should be zero. Nice catch :clap: and thank you. That's something I must add more unit tests tomorrow.

Here are some more quick tests, it seems that `.toString()` messed things up, also using a shadow conversion like postfixing `n` doesn't seem to work as expected.
```typescript
> const x = 0x7fffffffffffffff
undefined
> typeof(x)
'number'
> BigInt(x.toString()) - 0x7fffffffffffffffn
193n
> BigInt(x) - 0x7fffffffffffffffn
1n
> BigInt(x) - BigInt(0x7fffffffffffffff)
0n
> BigInt(x) - BigInt(9223372036854776000)
0n
```

Do you have an idea how this can be completely fixed? I think removing `.toString()` is enough but that will be confirmed when I push the unit tests covering wide-bit range numbers.

Reply:
Makes sense. Will try to refactor it in an elegant way and once I'm done will ping you again for review.

Reply:
What would be the advantage of using `json-bigint` over `bigint`? Only parsing? doesn't `bigint` already comes with parsing capabilities?

Reply:
`json-bigint` is doing `JSONBigInt.parse(data)` but under the hood, it transforms numeric values to `bigint` instead of JS Number.

If we're doing `JSON.parse()` and then manually transforming numeric values to Number, `JSON.parse()` already rounded the number and we lost it.

`json-bigint` doesn't replace `bigint` and they are not correlated. `json-big-int` is used instead of `JSON.parse()` due to its capability of converting string json input directly to `bigint` instead of JS Number.

Reply:
Thanks for the clarification.